Sources/LLBuildManifest/ManifestWriter.swift

```diff
@@ -11,63 +11,108 @@
 //===----------------------------------------------------------------------===//
 
 import Basics
-import protocol TSCBasic.ByteStreamable
-import struct TSCBasic.Format
-import protocol TSCBasic.OutputByteStream
-import class TSCBasic.BufferedOutputByteStream
+
+private let namesToExclude = [".git", ".build"]
 
 public struct ManifestWriter {
-    let fileSystem: FileSystem
+    private let manifest: BuildManifest
+    private var buffer = """
+    client:
+      name: basic
+    tools: {}
+
+    """
+
+    init(manifest: BuildManifest, buffer: String = """
+    client:
+      name: basic
+    tools: {}
+
+    """) {
+        self.manifest = manifest
+        self.buffer = buffer
+
+        self.render(targets: manifest.targets)
+
+        self.buffer += "default: \(manifest.defaultTarget.asJSON)\n"
 
-    public init(fileSystem: FileSystem) {
-        self.fileSystem = fileSystem
+        self.render(nodes: manifest.commands.values.flatMap { $0.tool.inputs + $0.tool.outputs })
+
+        self.render(commands: manifest.commands)
     }
 
-    public func write(
-        _ manifest: BuildManifest,
-        at path: AbsolutePath
-    ) throws {
-        let stream = BufferedOutputByteStream()
-        stream.send(
-            """
-            client:
-              name: basic
-            tools: {}
-            targets:\n
-            """
-        )
+    public static func write(_ manifest: BuildManifest, at path: AbsolutePath, _ fileSystem: FileSystem) throws {
+        let writer = ManifestWriter(manifest: manifest)
+
+        try fileSystem.writeFileContents(path, string: writer.buffer)
+    }
 
-        for (_, target) in manifest.targets.sorted(by: { $0.key < $1.key }) {
-            stream.send("  ").send(Format.asJSON(target.name))
-            stream.send(": ").send(Format.asJSON(target.nodes.map(\.name).sorted())).send("\n")
+    private mutating func render(targets: [BuildManifest.TargetName: Target]) {
+        self.buffer += "targets:\n"
+        for (_, target) in targets.sorted(by: { $0.key < $1.key }) {
+            self.buffer += "  \(target.name.asJSON): \(target.nodes.map(\.name).sorted().asJSON)\n"
         }
+    }
+
+    private mutating func render(nodes: [Node]) {
+        // We need to explicitly configure certain kinds of nodes.
+        let directoryStructureNodes = Set(nodes.filter { $0.kind == .directoryStructure })
```

Comment:
It's probably fine to separate all these for now, but it is valid for a node to be, for example, both a directory structure and mutating node